### ISSUE TITLE
KeyError in state_forwards During 0005_auto_20160727_2333.py Migration

### ISSUE THREAD

**Issue by jeffrey-hogan** [python-social-auth/social-app-django#8]
When migrating from python-social-auth to the new Django-specific app, the following exception is being thrown when running migrations:

```
$ python manage.py migrate social_django
Operations to perform:
  Apply all migrations: social_django
Running migrations:
  Applying social_django.0005_auto_20160727_2333...Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/vagrant/venv/lib/python3.4/site-packages/django/core/management/__init__.py", line 367, in execute_from_command_line
    utility.execute()
  File "/vagrant/venv/lib/python3.4/site-packages/django/core/management/__init__.py", line 359, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/vagrant/venv/lib/python3.4/site-packages/django/core/management/base.py", line 294, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/vagrant/venv/lib/python3.4/site-packages/django/core/management/base.py", line 345, in execute
    output = self.handle(*args, **options)
  File "/vagrant/venv/lib/python3.4/site-packages/django/core/management/commands/migrate.py", line 204, in handle
    fake_initial=fake_initial,
  File "/vagrant/venv/lib/python3.4/site-packages/django/db/migrations/executor.py", line 115, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/vagrant/venv/lib/python3.4/site-packages/django/db/migrations/executor.py", line 145, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/vagrant/venv/lib/python3.4/site-packages/django/db/migrations/executor.py", line 244, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/vagrant/venv/lib/python3.4/site-packages/django/db/migrations/migration.py", line 119, in apply
    operation.state_forwards(self.app_label, project_state)
  File "/vagrant/venv/lib/python3.4/site-packages/django/db/migrations/operations/models.py", line 521, in state_forwards
    model_state = state.models[app_label, self.name_lower]
KeyError: ('social_django', 'association')
```

Current state of the associated social_auth migrations and existing tables:

```
$ psql dev -c '\d' | grep social_auth
 public | social_auth_association                                         | table    | vagrant
 public | social_auth_association_id_seq                                  | sequence | vagrant
 public | social_auth_code                                                | table    | vagrant
 public | social_auth_code_id_seq                                         | sequence | vagrant
 public | social_auth_nonce                                               | table    | vagrant
 public | social_auth_nonce_id_seq                                        | sequence | vagrant
 public | social_auth_usersocialauth                                      | table    | vagrant
 public | social_auth_usersocialauth_id_seq                               | sequence | vagrant
```
```
$ psql dev -c "SELECT * FROM django_migrations WHERE app = 'social_auth'"
 id  |     app     |            name             |            applied
-----+-------------+-----------------------------+-------------------------------
 118 | social_auth | 0005_auto_20160727_2333     | 2016-08-15 15:41:28.565237+00
 119 | social_auth | 0003_alter_email_max_length | 2016-08-15 15:41:29.247433+00
 120 | social_auth | 0004_auto_20160423_0400     | 2016-08-15 15:41:29.249949+00
 121 | social_auth | 0001_initial                | 2016-08-15 15:41:29.251842+00
 122 | social_auth | 0002_add_related_name       | 2016-08-15 15:41:29.253419+00
(5 rows)
```

**Comment by hartym** [comment]
I have the exact same problem. Of course, not locally, only when trying to get it on production containers :) Let me know if there are details needed that may help in fixing the issue.

**Comment by omab** [comment]
I looks like [0004_auto_20160423_0400.py](https://github.com/python-social-auth/social-app-django/blob/master/social_django/migrations/0004_auto_20160423_0400.py) is not getting applied, that should setup the replacement name for `social_auth.association` to `social_django.association`.

**Comment by hartym** [comment]
Ok.

Following your exaplanation, I tryed to finish my release this morning and I succeeded.

What happened is that, for a reason I don't explain, the tables from social_auth existed in database, but the social_auth migrations were not marked as "applied" in the migrations table. I deployed a container with a pre-social-auth-migration version of the site, applied the migrations, then deployed the new version and now the migrations are applying.

I guess that maybe, even if it sounds obvious now that I say it, something should say in the docs to make sure that all social auth migrations are applied before switching to the new packages.

Thanks for your help!
